test/integration/test_metadata_strategy.py

```diff
@@ -0,0 +1,30 @@
+from galaxy_test.base.populators import (
+    DatasetPopulator,
+    uses_test_history,
+)
+from galaxy_test.driver.integration_util import IntegrationTestCase
+
+
+class TestDiskUsageUpdateDefault(IntegrationTestCase):
+    framework_tool_and_types = True
+
+    def setUp(self):
+        super().setUp()
+        self.dataset_populator = DatasetPopulator(self.galaxy_interactor)
+
+    @uses_test_history(require_new=False)
+    def test_run_work_dir_glob(self, history_id):
+        # Run a tool with a work dir glob and ensure content and disk usage is updated.
+        self.dataset_populator.new_dataset(history_id, content="fwd1Test", wait=True)
+        initial_disk_usage = self.dataset_populator.total_disk_usage()
+        response = self.dataset_populator.run_tool("from_work_dir_glob", {}, history_id, assert_ok=True)
+        self.dataset_populator.wait_for_job(job_id=response["jobs"][0]["id"])
+        final_disk_usage = self.dataset_populator.total_disk_usage()
+        assert final_disk_usage == initial_disk_usage + 3
+
+
+class TestDiskUsageCeleryExtended(TestDiskUsageUpdateDefault):
+    @classmethod
+    def handle_galaxy_config_kwds(cls, config):
+        super().handle_galaxy_config_kwds(config)
+        config["metadata_strategy"] = "celery_extended"
```

Comment:
I was just trying this, as suggested by Marius, to see if this was somehow related to a possible issue when using Celery that I'm currently investigating (#15821).

What I found out about this is that when the `metadata_strategy` is set to `directory` or `directory_celery` both test success, and they fail with `extended` and `extended_celery` so it seems that Celery is not the cause of this problem but something with the `extended` metadata strategy itself.